K blocker rc

BruceK4t1qbit opened the pull request:
Changed the k-blocker observations from using a single integer to encode the position of each block to using 2 integers (the x and y coordinates) to encode the position of each block.

Changed tests accordingly, fixed one test, and created one new test for a bug I created.

BruceK4t1qbit committed:
Refactored test_kblocker:

Extracted environment setup boilerplate from all tests to a function.

BruceK4t1qbit committed:
Refactored kblocker:

Extracted a "self._get_encoded_state()" method.

BruceK4t1qbit committed:
Changed kblocker.py so that the state is now the x and y coordinates of all states

Previously, the position of each piece was one-hot encoded.

Also, updated the tests accordingly, fixed one test, and created a new test to catch an error in mismatched dimensions that I initially created while I was changing kblocker.py.

majiddadashi commented on gym/envs/nysaq/tests/assets/rbmConfig1bt:
```diff
@@ -24,7 +24,7 @@
     "grid_y": 5,
-    "positive_reward": 1,
-    "negative_reward": 0
+    "positive_reward APPARENTLY NOT USED": 1,
```
Can we actually use these instead? we can rename them:
win_reward: For winning. currently 5.
play_reward: For taking each action. currently 0.

BruceK4t1qbit replied:
Implemented

majiddadashi commented on gym/envs/nysaq/kblocker.py:
```diff
@@ -105,15 +107,19 @@ def _reset(self):
+
+
+    def _get_encoded_state(self):
+        return sum(map(list, self.__coords), [])
```
Could you please add a comment here for readability? took me a few minutes to figure out what's happening here.

BruceK4t1qbit replied:
Done.  And refactored the code so it is much simpler!

BruceK4t1qbit committed:
Changed kblocker.py so that the win reward and move rewards are loaded from config files.

Changed config files and test file accordingly.

Also, cleaned up and documented "_get_encoded_state" method
